pkg/cloud/gcp/gcs_storage.go

```diff
@@ -124,18 +139,44 @@ func makeGCSStorage(
 		// Do nothing; use implicit params:
 		// https://godoc.org/golang.org/x/oauth2/google#FindDefaultCredentials
 	default:
-		if conf.Credentials == "" {
-			return nil, errors.Errorf(
-				"%s must be set if %q is %q",
-				CredentialsParam,
-				cloud.AuthParam,
-				cloud.AuthParamSpecified,
-			)
-		}
-
-		authOption, err := createAuthOption(conf.Credentials)
-		if err != nil {
-			return nil, errors.Wrapf(err, "error getting credentials from %s", CredentialsParam)
+		var authOption option.ClientOption
+		var err error
+
```

Comment:
can we restructure the if condition to reduce nesting. imo something like this is more readable?

```
if  conf.TemporaryToken != "" {
   if !temporaryTokenEnabled.Get(&args.Settings.SV) {
       // setting disables use of temp token
   }
   // temp logic
} else if conf.Credentials != "" {
  // creds logic
} else {
 // invalid param, only specify one of cred/token.
}
```

Reply:
done. feature flag is removed.